src/util/user-dirs.js

```diff
@@ -0,0 +1,45 @@
+/* @flow */
+
+const path = require('path');
+const userHome = require('./user-home-dir').default;
+
+export function getDataDir(): string {
+  if (process.platform === 'win32') {
+    return path.join(getLocalAppDataDir(), 'Data');
+  } else if (process.env.XDG_DATA_HOME) {
+    return path.join(process.env.XDG_DATA_HOME, 'yarn');
+  } else {
+    // This could arguably be ~/Library/Application Support/Yarn on Macs,
+    // but that feels unintuitive for a cli tool
+
+    // Instead, always use the typical location of XDG_DATA_HOME
+    return path.join(userHome, '.local', 'share', 'yarn');
+  }
+}
+
+export function getCacheDir(): string {
+  if (process.platform === 'win32') {
+    // process.env.TEMP also exists, but most apps put caches here
```

Comment:
@Daniel15 do you know about this?

Reply:
TEMP is for temp files (anything you don't care about any more after some period of time while it's being used), whereas I think we want to keep the Yarn cache around.